kartothek/io/dask/_update.py

```diff
@@ -307,14 +308,33 @@ def update_dask_partitions_one_to_one(
     if secondary_indices:
         mps = map_delayed(MetaPartition.build_indices, mps, columns=secondary_indices)
 
-    return map_delayed(
+    mps = map_delayed(
         MetaPartition.store_dataframes,
         mps,
         store=store_factory,
         df_serializer=df_serializer,
         dataset_uuid=dataset_uuid,
     )
 
+    if metadata_storage_factory:
+
+        def send_mp_to_queue(mp, metadata_storage_factory):
+            # We can't msgpack serialize Metapartition by itself, so we serialize each mp dict
+            # We need to instantiate a worker client to be able to push to a queue
+            from distributed import get_client
+
+            c = get_client()  # noqa
+            metadata_store = metadata_storage_factory()
+            for _mp in mp.metapartitions:
+                metadata_store.put(_mp)
```

Comment:
this task might get retried. Do you do any de-duplication of the results in the end?

Reply:
good point

- [ ] make `partition_list` a set in `update_dataset_from_partitions` when `metadata_storage_factory`, to prevent potential duplicates